account_credit_control/models/credit_control_communication.py

```diff
@@ -5,59 +5,59 @@
 from odoo import api, fields, models
 
 
-class CreditControlCommunication(models.TransientModel):
-    """Shell class used to provide a base model to email template and reporting
-    If used this approach in version 7 a browse record
-    will exist even if not saved
-
-    """
+class CreditControlCommunication(models.Model):
     _name = "credit.control.communication"
-    _description = "credit control communication"
+    _inherit = ["mail.thread", "mail.activity.mixin"]
+    _description = "Credit control communication process"
     _rec_name = 'partner_id'
 
     partner_id = fields.Many2one(
         comodel_name='res.partner',
         string='Partner',
         required=True,
+        index=True,
+    )
+    policy_id = fields.Many2one(
+        related='policy_level_id.policy_id',
+        store=True,
+        index=True,
     )
-    current_policy_level = fields.Many2one(
+    policy_level_id = fields.Many2one(
         comodel_name='credit.control.policy.level',
         string='Level',
         required=True,
+        index=True,
     )
     currency_id = fields.Many2one(
         comodel_name='res.currency',
         string='Currency',
         required=True,
+        index=True,
     )
-    credit_control_line_ids = fields.Many2many(
+    credit_control_line_ids = fields.One2many(
         comodel_name='credit.control.line',
-        rel='comm_credit_rel',
+        inverse_name="communication_id",
         string='Credit Lines',
     )
-    contact_address = fields.Many2one(
+    contact_address_id = fields.Many2one(
```

Comment:
There's a chance that the records on the table are not vacuum, so better to rename these fields in scripts for not losing references. Shouldn't this field be required by the way?

Reply:
> There's a chance that the records on the table are not vacuum, so better to rename these fields in scripts for not losing references

Good catch, but actually in that case, this table would become problematic, because it would contain data that was not meant to be persistent. What I'm going to do is force a vacuum in the pre-migration.

> Shouldn't this field be required by the way?

It's used automatically when the action is executed, and in those cases is always filled automatically. From then, the only thing that actually matters is message followers, so I think it is OK as it is.